src/Microsoft.ComponentDetection.Contracts/TypedComponent/CondaLockComponent.cs

```diff
@@ -0,0 +1,47 @@
+namespace Microsoft.ComponentDetection.Contracts.TypedComponent;
+
+using System.Collections.Generic;
+
+public class CondaLockComponent : TypedComponent
+{
+    public CondaLockComponent(string name, string version, string category, Dictionary<string, string> dependencies, Dictionary<string, string> hash, string manager, bool optional, string platform, string url)
+    {
+        this.Name = this.ValidateRequiredInput(name, nameof(this.Name), nameof(ComponentType.Conda));
+        this.Version = this.ValidateRequiredInput(version, nameof(this.Version), nameof(ComponentType.Conda));
+        this.Category = category;
+        this.Dependencies = dependencies;
+        this.Hash = hash;
+        this.Manager = manager;
+        this.Optional = optional;
+        this.Platform = platform;
+        this.Url = url;
+        this.Version = version;
+    }
+
+    private CondaLockComponent()
+    {
+        /* Reserved for deserialization */
+    }
+
+    public string Category { get; set; }
+
+    public Dictionary<string, string> Dependencies { get; set; }
+
+    public Dictionary<string, string> Hash { get; set; }
+
+    public string Manager { get; set; }
+
+    public string Name { get; set; }
+
+    public bool Optional { get; set; }
+
+    public string Platform { get; set; }
```

Comment:
Are any of these equivalent: Category, Build, Channel, Manager and Namespace? I see you used very different property names as the existing Conda component we had, are these new ones lockfile specific? We should use the most generic terminology applicable for Conda.

Reply:
I refactored the code to use the existing CondaComponent. So this should be resolved.